src/phylum/ci/manifest.py

```diff
@@ -0,0 +1,137 @@
+"""Define an implementation for manifests.
+
+A manifest contains project dependencies in their loose, unresolved form.
+"""
+
+from functools import cached_property
+import json
+from pathlib import Path
+import shlex
+import subprocess
+import tempfile
+import textwrap
+
+from phylum.ci.common import PackageDescriptor, Packages
+from phylum.ci.depfile import Depfile, parse_current_depfile
+from phylum.exceptions import PhylumCalledProcessError, pprint_subprocess_error
+from phylum.logger import LOG, MARKUP
+
+
+class Manifest(Depfile):
+    """Provide methods for operating on a manifest."""
+
+    @cached_property
+    def current_deps(self) -> Packages:
+        """Get the current manifest packages and return them in sorted order."""
+        try:
+            curr_manifest_packages = parse_current_depfile(self.cli_path, self.type, self.path)
+        except subprocess.CalledProcessError as err:
+            msg = f"""\
+                Consider supplying lockfile type explicitly in the `.phylum_project` file.
+                  For more info, see: https://docs.phylum.io/docs/lockfile_generation
+                  Please report this as a bug if you believe [code]{self!r}[/]
+                  is a valid [code]{self.type}[/] manifest file."""
+            raise PhylumCalledProcessError(err, textwrap.dedent(msg)) from err
+        return sorted(set(curr_manifest_packages))
+
+    @cached_property
+    def base_deps(self) -> Packages:
+        """Get the dependencies from the base iteration of the manifest and return them in sorted order.
+
+        The base iteration is determined by the common ancestor commit.
+        """
+        if not self.common_ancestor_commit:
+            LOG.info("No common ancestor commit for `%r`. Assuming no base dependencies.", self)
+            return []
+
+        with tempfile.TemporaryDirectory(prefix="phylum_") as temp_dir:
+            cmd = ["git", "worktree", "add", "--detach", temp_dir, self.common_ancestor_commit]
+            LOG.debug("Adding a git worktree for the base iteration in a temporary directory ...")
+            LOG.debug("Using command: %s", shlex.join(cmd))
+            try:
+                subprocess.run(cmd, check=True, capture_output=True, text=True)  # noqa: S603
```

Comment:
Creating this file structure for each individual manifest file seems pretty inefficient... Can this code be restructured to work on a list of depfiles so that the working tree can be reused?

Reply:
Maybe...an attempt will be made.

Reply:
Update...this restructuring will still be attempted...but possibly in a different PR...or at least a different commit. Proceeding with a review of this branch in it's current state is desired.

Reply:
The change was possible. This PR has been updated to reflect it.